src/chain.ts

```diff
@@ -181,6 +193,8 @@ export const chain = (
 
         // Same here.
         sourceSocket.on('close', () => {
+            sourceSocket.unpipe(targetSocket);
+            targetSocket.unpipe(sourceSocket);
```

Comment:
Can we comment on why is this needed? There are some more places with this as well.